From 9a9b098a1963687a03a005ac10684b4cff16614e Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Fri, 27 Nov 2015 11:22:32 +0800
Subject: [PATCH 0705/5242] MLK-11915-09 ASoC: cs42xx8: remove the no effect
 comparison

commit  8a1bfd947e89f0bbd771304c84cbe3f0da3eaaed from
https://source.codeaurora.org/external/imx/linux-imx.git

rate_tx and rate_rx are always greater than or equal to 0, so remove the
no effect comparison. Reported by Coverity.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit d5957250784a27cf6f6f421d8a6e0ba45b4bbc18)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/cs42xx8.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/cs42xx8.c b/sound/soc/codecs/cs42xx8.c
index 803a89a..900808a 100644
--- a/sound/soc/codecs/cs42xx8.c
+++ b/sound/soc/codecs/cs42xx8.c
@@ -264,7 +264,7 @@ static int cs42xx8_hw_params(struct snd_pcm_substream *substream,
 		fm_rx = CS42XX8_FM_AUTO;
 		fm_tx = CS42XX8_FM_AUTO;
 	} else {
-		if (rate_tx >= 0 && rate_tx < 50000)
+		if (rate_tx < 50000)
 			fm_tx = CS42XX8_FM_SINGLE;
 		else if (rate_tx > 50000 && rate_tx < 100000)
 			fm_tx = CS42XX8_FM_DOUBLE;
@@ -275,7 +275,7 @@ static int cs42xx8_hw_params(struct snd_pcm_substream *substream,
 			return -EINVAL;
 		}
 
-		if (rate_rx >= 0 && rate_rx < 50000)
+		if (rate_rx < 50000)
 			fm_rx = CS42XX8_FM_SINGLE;
 		else if (rate_rx > 50000 && rate_rx < 100000)
 			fm_rx = CS42XX8_FM_DOUBLE;
-- 
1.7.9.5

